src/main/java/com/artipie/rpm/meta/XmlMetaJoin.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.rpm.meta;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import javax.xml.stream.XMLEventFactory;
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLEventWriter;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLOutputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.events.XMLEvent;
+
+/**
+ * Joins two meta xml-files.
+ * @since 0.9
+ */
+public final class XmlMetaJoin {
+
+    /**
+     * Tag.
+     */
+    private final String tag;
+
+    /**
+     * Ctor.
+     * @param tag Metatag
+     */
+    public XmlMetaJoin(final String tag) {
+        this.tag = tag;
+    }
+
+    /**
+     * Appends data from part to target.
+     * @param target Target
+     * @param part File to append
+     * @throws IOException On error
+     */
+    public void merge(final Path target, final Path part) throws IOException {
+        final Path res = target.getParent().resolve(
+            String.format("%s.joined", target.getFileName().toString())
+        );
+        try (OutputStream out = Files.newOutputStream(res)) {
+            final XMLEventWriter writer = XMLOutputFactory.newInstance().createXMLEventWriter(out);
```

Comment:
@olenagerasimova can we put this `writer` to the same try with resources expression as `out`?
```java
try (OutputStream out = ...;
    XMLEventWriter writer = ...) {
}
```

Reply:
@g4s8 no, unfortunately, `XMLEventWriter` does not implement `AutoCloseable`